### PR TITLE
More careful code for all of `-x` mode

### PR DESCRIPTION
**Pull request by mxcl**
not merging until coverage on these pieces is higher.

Shame we have no automated code-coverage :(

### CODE REVIEW

File: src/app.exec.ts [hunk 1]
```diff
@@ -197,7 +197,7 @@ async function repl(installations: Installation[], env: Record<string, string>)
   const shell = Deno.env.get("SHELL")?.trim() || "/bin/sh"
   const cmd = [shell, '-i'] // interactive
 
-  //TODO other shells pls #help-wanted
+  //TODO other shells pls #help-wlanted
```
Comment (jhheider) [inline src/app.exec.ts:200]: um?

**Comment by mxcl** [comment]
k well, writing the tests is daunting. Needs a lot of infra and I have to move on to other things for momentum reasons.

yolo?

**Comment by jhheider** [comment]
Yes, I think until we have a much broader base, there will be some yoloing of major overhauls. Since they markedly increase quality, I believe we will be forgiven for minor hits to stability along the way.

_for now._

File: src/app.exec.ts (hunk 1 above)
Comment (mxcl) [reply src/app.exec.ts:222]: lol

**Pull request merged**